apps/req.c

```diff
@@ -803,9 +818,11 @@ int req_main(int argc, char **argv)
             if (!X509_set_issuer_name(new_x509, issuer))
                 goto end;
             if (days == UNSET_DAYS) {
-                days = DEFAULT_DAYS;
-            }
-            if (!set_cert_times(new_x509, NULL, NULL, days))
+                if (not_after == NULL)
+                    days = DEFAULT_DAYS;
```

Comment:
Again, I do not see why you place the assignment under the condition `not_after == NULL`, but anyway.

Reply:
Agree :+1:
I think I had a knot in my brain regarding that one.

Reply:
This proposal is applied.